celery/utils/functional.py

```diff
@@ -241,12 +241,18 @@ def __bool__(self):
 
     @property
     def data(self):
-        try:
+        if not self.__done:
             self.__consumed.extend(list(self.__it))
```

Comment:
An intermediate `list()` should not be needed. I.e. `self.__consumed.extend(self.__it)` should do.

Reply:
changed